File: API/Models/Flight.cs
using System;

namespace API.Models;

public class Flight
{
    public int Id { get; set; }
    public DateTime DepartureTime { get; set; }
    public DateTime ArrivalTime { get; set; }
    public int Duration { get; set; }
    public required string Status { get; set; }
    public decimal Price { get; set; }
    public int AvailableSeats { get; set; }
    public List<Booking> Bookings { get; set; } = [];

    public int AircraftId { get; set; }
    public Aircraft Aircraft { get; set; } = null!;
    public int DepartureAirportId { get; set; }
    public Airport DepartureAirport { get; set; } = null!;
    public int ArrivalAirportId { get; set; }
    public Airport ArrivalAirport { get; set; } = null!;

}

File: client/src/models/flight.ts
export interface Flight {
  id: number;
  departureTime: Date;
  arrivalTime: Date;
  price: number;
  availableSeats: number;
  duration: number;
  status: string;
  departureAirportName: string;
  arrivalAirportName: string;
  aircraftType: string;
}

export interface AdminFlight extends Flight {
  aircraftId: number;
  departureAirportId: number;
  arrivalAirportId: number;
}

export interface FlightEditorRequest {
  departureTime: string;
  arrivalTime: string;
  price: number;
  availableSeats: number;
  status: string;
  aircraftId: number;
  departureAirportId: number;
  arrivalAirportId: number;
}

export interface DeleteFlightResult {
  succeeded: boolean;
  flightId: number;
  message: string;
  errors: string[];
}

export interface Airport {
  id: number;
  name: string;
  city: string;
  country: string;
  iataCode: string;
}

export interface SearchFlightRequest {
  departureAirportId: number;
  arrivalAirportId: number;
  departureTime: Date;
  numberOfPassengers: number;
  sortBy?: string;
}

export interface AircraftOption {
  id: number;
  model: string;
  registrationNumber?: string;
  totalSeats: number;
  seatRows: number;
  seatsPerRow: number;
}
